packages/api/test/functional/technology-cost.spec.js

```diff
@@ -211,3 +223,65 @@ test('PUT /technologies/:id/costs deletes costs with empty cost array.', async (
 		costs: [],
 	});
 });
+
+test('PUT /technologies/:id/costs calls algoliasearch.saveObject with implementation and maintenance costs if it is provided', async ({
+	assert,
+	client,
+}) => {
+	const defaultTermFem = 'Não definida';
+	const defaultTermMasc = 'Não definido';
+
+	const lastTechnology = await Technology.last();
+	const technologyCostInst = await lastTechnology.technologyCosts().first();
+
+	const loggeduser = await User.create(researcherUser);
+	await lastTechnology.users().attach([loggeduser.id]);
+
+	await technologyCostInst.load('costs');
+
+	const updatedTechnologyCost = technologyCostInst.toJSON();
+
+	const newImplementationCost = {
+		cost_type: 'implementation_costs',
+		description: 'Custo de implantação adicional',
+		type: 'material',
+		quantity: 2,
+		value: 10000,
+	};
+
+	const newMaintenanceCost = {
+		cost_type: 'maintenance_costs',
+		description: 'Custo de manutenção adicional',
+		type: 'material',
+		quantity: 1,
+		value: 500,
+	};
+
+	updatedTechnologyCost.costs = [];
+	updatedTechnologyCost.costs.push(newImplementationCost);
+	updatedTechnologyCost.costs.push(newMaintenanceCost);
+
+	const response = await client
+		.put(`/technologies/${lastTechnology.id}/costs`)
+		.loginVia(loggeduser, 'jwt')
+		.send(updatedTechnologyCost)
+		.end();
+
+	const updatedTechnology = await Technology.find(response.body.id);
+	await updatedTechnology.load('users');
+
+	assert.isTrue(AlgoliaSearch.initIndex.called);
+	assert.isTrue(
+		AlgoliaSearch.initIndex().saveObject.withArgs({
+			...updatedTechnology.toJSON(),
+			category: defaultTermFem,
+			classification: defaultTermFem,
+			dimension: defaultTermFem,
+			targetAudience: defaultTermMasc,
+			implementationCost: 20000,
+			maintenanceCost: 500,
+			institution: loggeduser.company,
+			thumbnail: null,
+		}).calledOnce,
```

Comment:
I hate that we have to do this for each new term 😄 .

Reply:
Yeah, unfortunately this is not a pleasant job 😅